zerver/views/invite.py

```diff
@@ -232,6 +233,55 @@ def resend_user_invite_email(
     return json_success(request)
 
 
+@require_member_or_admin
+@typed_endpoint
+def edit_multiuse_invite(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    *,
+    invite_id: PathOnly[int],
+    stream_ids: Json[list[int]] | None = None,
+    invite_as: Annotated[
+        Json[int] | None,
+        check_int_in_validator(list(PreregistrationUser.INVITE_AS.values())),
+    ] = None,
+    include_realm_default_subscriptions: Json[bool] = False,
+) -> HttpResponse:
+    try:
+        invite = MultiuseInvite.objects.get(id=invite_id)
+    except MultiuseInvite.DoesNotExist:
+        raise JsonableError(
+            _("Invite does not exist with id: {invite_id}.").format(invite_id=invite_id)
+        )
+
+    if (
+        invite.referred_by_id != user_profile.id
+        and not user_profile.is_realm_admin
+        and not user_profile.can_create_multiuse_invite_to_realm()
+    ):
+        # Guest users case will not be handled here as it will
+        # be handled by the decorator above.
+        raise JsonableError(_("Insufficient permission"))
+
+    if invite_as is not None:
+        require_admin = invite_as in [
+            PreregistrationUser.INVITE_AS["REALM_OWNER"],
+            PreregistrationUser.INVITE_AS["REALM_ADMIN"],
+            PreregistrationUser.INVITE_AS["MODERATOR"],
+        ]
+        check_role_based_permissions(invite.invited_as, user_profile, require_admin=require_admin)
+
+    streams: list[Stream] = []
```

Comment:
I missed this before, but I guess this would set the streams to empty if `stream_ids` was passed as `None` to request meaning that user did not intended to change the value.

Reply:
There can be two ways to solve this - we can either set the streams to original set of streams if it is passed as `None` in api request or we can just pass it as `None` and update `do_edit_multiuse_invite_link` to set `streams` field only if it is not `None`.